### PR TITLE
Client representation on backend.

### PR DESCRIPTION
**Pull request by LudwigVonChesterfield**
- Client class represents the current connection with some client.
- Client_info class represents the info that should be preserved between different clients of same *ownership*.

Works on, but doesn't close yet: https://trello.com/c/KgZ4wVE2/39-client-class-representing-client-on-backend#

### CODE REVIEW

**Comment by LudwigVonChesterfield** [comment]
[Client class, representing client on backend.](https://trello.com/c/KgZ4wVE2/39-client-class-representing-client-on-backend)

File: code/client.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        pass
+
+    @property
+    def email(self):
+        return self._email
+
+    @email.setter
+    def email(self, new_email):
+        self._email = new_email
+        if new_email != self._email:
```
Comment (lynnporu) [inline code/client.py]: Will this condition be true for ever?

File: code/client.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    session_obj is who they pretend to be.
+
+    Returns None otherwise.
+    """
+    """
+    user_id = session_obj.get("user_id")
+    key = session_obj.get("key")
+    salt = session_obj.get("salt")
+
+    if(user_id is not None and key is not None and salt is not None):
```
Comment (lynnporu) [inline code/client.py]:
```suggestion
    if all(user_id, key, salt):
```

File: code/client.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+class Client_Info:
+    def __init__(self, user_id=USER_ID_ANONYMOUS, ip=""):
+        self.user_id = user_id
+
+        # These we will get from the db.
+        self._email = ""
+        self._discord_id = ""
+        self._permissions = SERVER_PERMISSION_BASIC
+
+        global_vars.client_infos_by_user_id[str(user_id)] = self
```
[the remark this answers is not in the document]
Comment (LudwigVonChesterfield) [reply code/client.py]: well, it is a lookup table after all

**Comment by LudwigVonChesterfield** [comment]
self merge bad

**Pull request merged**